### DISCUSSION TITLE
Stable dependency versions

### DISCUSSION THREAD

**Discussion by RamDivrania** [NeoSOFT-Technologies/rest-node-nestjs#8] [Ideas]
Instead of allowing auto-update for dev or prod dependencies, we should set it to fixed/stable versions and only allow manual updates/upgrades. Or we should allow auto-update only for patch versions.
In many cases, dependencies can break builds.

~~`"typescript": "^4.2.3"`~~

- [ ] `"typescript": "4.2.3"`

- [ ] `"typescript": "~4.2.3"`

**Comment by sasachan** [comment]
Yes,

As per stable product development practice production and pre-prod(stage) auto-upgrade should be locked, However, for development environment it should be auto-upgrade. 

@RamDivrania Good suggestion.

CC: @santoshshinde2012
